c-dependencies/js-compute-runtime/builtins/text-encoder.cpp

```diff
@@ -0,0 +1,81 @@
+#include "builtins/text-encoder.h"
+
+#include "js/ArrayBuffer.h"
+
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#include "js/experimental/TypedData.h"
+#pragma clang diagnostic pop
+
+namespace builtins {
+
+bool TextEncoder::encode(JSContext *cx, unsigned argc, JS::Value *vp) {
+  METHOD_HEADER(1)
+
+  // Default to empty string if no input is given.
+  if (args.get(0).isUndefined()) {
+    JS::RootedObject byte_array(cx, JS_NewUint8Array(cx, 0));
+    if (!byte_array)
+      return false;
+
+    args.rval().setObject(*byte_array);
+    return true;
+  }
+
+  size_t chars_len;
+  JS::UniqueChars chars = ::encode(cx, args[0], &chars_len);
+
+  auto *rawChars = chars.release();
+  JS::RootedObject buffer(cx, JS::NewArrayBufferWithContents(cx, chars_len, rawChars));
+  if (!buffer) {
+    JS_free(cx, rawChars);
+    return false;
+  }
+
+  JS::RootedObject byte_array(cx, JS_NewUint8ArrayWithBuffer(cx, buffer, 0, chars_len));
+  if (!byte_array)
+    return false;
+
+  args.rval().setObject(*byte_array);
+  return true;
+}
+
+bool TextEncoder::encoding_get(JSContext *cx, unsigned argc, JS::Value *vp) {
+  METHOD_HEADER(0)
+
+  JS::RootedString str(cx, JS_NewStringCopyN(cx, "utf-8", 5));
```

Comment:
Non-blocking comment: Since this can never change, should we make this a persistent string instead of creating a new string each time? JS Strings are immutable, so a persistent string should work here

Reply:
I think that's a good change, but let's do it in a follow :+1: